Doc/whatsnew/3.14.rst

```diff
@@ -506,6 +506,16 @@ http
   module allow the browser to apply its default dark mode.
   (Contributed by Yorik Hansen in :gh:`123430`.)
 
+* The :mod:`http.server` module now supports serving over HTTPS using the
+  new :class:`http.server.HTTPSServer`. Furthermore, the following command-line
+  options have been added to ``python -m http.server``:
+
+  * ``--tls-cert <path>``: Path to the TLS certificate file.
```

Comment:
For a follow-up PR, I'd like to be able to have links to that. I think we can create an issue for adding links to all command-line programs (docs only). It would be helpful for modules where the command-line interface is burried in a huge page.